contracts/Depository.sol

```diff
@@ -305,8 +358,16 @@ contract Depository is IErrorsTokenomics {
             revert ProductClosed(productId);
         }
 
+        uint256 productMaxVestingTime = product.vesting;
+        // Calculate vesting limits
+        if (bondVestingTime < MIN_VESTING) {
+            revert LowerThan(bondVestingTime, MIN_VESTING);
+        }
+        if (bondVestingTime > productMaxVestingTime) {
+            revert Overflow(bondVestingTime, productMaxVestingTime);
+        }
```

Comment:
Bond vesting time is now chosen by the user from `MIN_VESTING` to max product defined vesting time.